test-cloud/test-cloud-common/src/main/java/org/kie/cloud/tests/common/client/util/WorkbenchUtils.java

```diff
@@ -65,6 +104,49 @@ private static CloneProjectRequest createCloneProjectRequest(String repositoryUr
         return cloneProjectRequest;
     }
 
+    private static void handleNotFoundException(WorkbenchClient workbenchClient, Exception originalException) {
+        logger.debug("Exception handled when workbench client throws 404 when is tried to search for all spaces.");
+        boolean workbenchWorks = false;
+        for (int i = 0; i < 10; i++) {
+            try {
+                if (workbenchClient.getSpaces().stream().map(Space::getName).anyMatch(SPACE_NAME::equals)) {
+                    logger.debug("Container URL works.");
+                    workbenchWorks = true;
+                    break;
+                } else {
+                    logger.debug("Container URL works, but space was not found. Waiting for 30 seconds and trying again.");
+                    waitFor30Seconds();
+                }
+            } catch (NotFoundException ee) {
+                logger.debug("Container URL do not work. Waiting for 30 seconds and trying again.", ee);
+                waitFor30Seconds();
+            }
+        }
+        if (workbenchWorks) {
+            logger.info("After few retries Workbench client works. NotFoundException can be ignored.");
+        } else {
+            throw new RuntimeException("Workbench client does not works", originalException);
+        }
+    }
+
+    private static void handleWaiterException(int tries, Exception originalException) {
+        if (tries < 4) {
+            logger.warn("Timeout while cloning project to the Workbench application.", originalException);
+        } else {
+            throw new RuntimeException("Timeout while cloning project to the Workbench application.",
+                                       originalException);
+        }
+    }
+
+    private static void waitFor30Seconds() {
+        try {
+            Thread.sleep(TimeUnit.SECONDS.toMillis(30));
+        } catch (InterruptedException e1) {
+            // TODO Auto-generated catch block
+            e1.printStackTrace();
```

Comment:
```suggestion
            Thread.currentThread().interrupt();
            throw new RuntimeException("Interrupted while waiting.", e1);
```

Reply:
Why did we need to either print or interrupt the thread? :/

Reply:
@jakubschwan The InterruptedException is thrown for example in case the process is getting terminated. In such case we should stop immediately and don't continue with execution flow.
@Sgitario I read somewhere that this is a standard handling of InterruptedException. So I prefer to follow it.